### PR TITLE
Allow to use local-ssd disk type

### LINKED ISSUE #768: Invalid equality prevents the usage of local-ssd disks
/kind bug

While trying to use local-ssd disks I came across the following error:

```bash
Error 400: Invalid value for field 'resource.disks[1].type
Cannot create local SSD as persistent disk
```

After further inspection of the source code it turns out that the following equality can never be satisfied.

https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/26add93f103a6ca8ca9fb9f26873cd5a51d64cf4/cloud/scope/machine.go#L257

I have a fix ready for it, however, before I open a pull request I have a question about unit tests.  
This seems to be a case where having unit tests could catch these classes of bugs.  
In my fix I did provide a unit test to validate that the fix is correct but, it seems, you are preventing to push any `_test.go` files, via`.gitignore`

Is this done on purpose ?

### PR DESCRIPTION
**Pull request by silenteh**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #768

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [x] includes documentation
- [x] adds unit tests

**Release note**:

```release-note
- Fixes a bug preventing local-ssd disks from being used.
```

### CODE REVIEW

**Comment by dims** [comment]
```
Running [/home/runner/golangci-lint-1.50.0-linux-amd64/golangci-lint run --out-format=github-actions] in [] ...
  Error: File is not `gci`-ed with --skip-generated -s standard,default (gci)
  Error: Comment should end in a period (godot)
  Error: unnecessary leading newline (whitespace)
  Error: unnecessary trailing newline (whitespace)
  
  level=warning msg="[linters_context] rowserrcheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649."
  
  Error: issues found
  Ran golangci-lint in 144[23](https://github.com/kubernetes-sigs/cluster-api-provider-gcp/actions/runs/3693383985/jobs/6257558870#step:4:25)5ms
```

**Comment by richardcase** [comment]
Thanks for this @silenteh. Would you be able to squash your commits?

File: cloud/scope/machine_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+// This test veries that if a user selects "local-ssd"
```
Comment (fiunchinho) [inline cloud/scope/machine_test.go:13]:
```suggestion
// This test verifies that if a user selects "local-ssd"
```

**Comment by silenteh** [comment]
> Thanks for this @silenteh. Would you be able to squash your commits?

Sure - I am just struggling to run the golangci-lint locally to check if all lint pass. Then I will squash the commits.

**Comment by richardcase** [comment]
> Sure - I am just struggling to run the golangci-lint locally to check if all lint pass. Then I will squash the commits.

Thanks @silenteh 🙇  What issues are you getting when running the linter locally? Feel free to ping me on slack if you need another pair of eyes.

File: cloud/scope/machine_test.go (hunk 1 above)
Comment (silenteh) [reply cloud/scope/machine_test.go]: Typo fixed

**Comment by silenteh** [comment]
> > Sure - I am just struggling to run the golangci-lint locally to check if all lint pass. Then I will squash the commits.
> 
> Thanks @silenteh 🙇 What issues are you getting when running the linter locally? Feel free to ping me on slack if you need another pair of eyes.

I am getting the following error:
```
can't run linter goanalysis_metalinter: goimports: can't extract issues from gofmt diff output 
```
I believe it's an issue in my local dev environment. Probably something to do with `goimports` and `go-diff` interoperability

**Comment by silenteh** [comment]
> > I believe it's an issue in my local dev environment. Probably something to do with `goimports` and `go-diff` interoperability
> 
> I had a similar issue which was caused by the version of `diff` on my machine. To fix it i did:
> 
> ```
> brew install diffutils
> ```

Yep that was the issue.  
Seems caused by the default `diff` command shipped with Apple which can't parse the timezone on dates.  
`https://github.com/golangci/golangci-lint/issues/3087`

Anyway...all linted and squashed.

**Pull request merged**